### DISCUSSION TITLE
Default BindableProperties and Automated Setter method for Two-Way Binding

### DISCUSSION THREAD

**Discussion by egvijayanand** [CommunityToolkit/Maui.Markup#343] [New Feature Discussions]
@brminnick If you recall, we discussed this a year ago #272.

At that time, I made changes to the Binding methods using Reflection.

Now, I've found a way to make it work without using Reflection.

This approach leverages **C# Expressions** that compile statically with no runtime overhead.

I'll modify the sample we used previously and update you on this.

This improvement could be a game changer, significantly enhancing developer productivity by providing the default setter method.

**Comment by egvijayanand** [comment]
@brminnick 

Have pushed the updated sample to GitHub. Kindly get the latest version.

https://github.com/egvijayanand/markup-issue-272

Check and let me know your feedback.

**Reply by TheCodeTraveler** [reply]
Hey Vijay! Is the idea to add more default bindings to `CommunityToolkit.Maui.Markup`? Or for default bindings to use TypedBindings instead of string-based bindings? Or both?

Personally, I don't like using default bindings because it hides the bindable property. But I'm totally cool if you want to add them! 

Could you open a [New Feature Proposal](https://github.com/CommunityToolkit/Maui.Markup/issues/new?template=open-a-new-feature-proposal.yml)? I'll bring it to our monthly standup on Thursday. And, if approved by the team, you'll be welcome to submit a PR for the code + unit tests in this repo and a PR in the [docs repo](https://github.com/MicrosoftDocs/CommunityToolkit/tree/main/docs/maui).

**Reply by egvijayanand** [reply]
Sure, will create a new feature proposal with the details.

**Reply by egvijayanand** [reply]
* #345

**Comment by egvijayanand** [comment]
This logic can also be applied to the existing methods, as many non-default bindable properties have support for two-way data binding.
